卸载区块时发生错误

# 🐞 错误汇报
<!-- 
👉 尽可能详实地描述错误信息
-->
报错 [ERROR] 卸载区块时发生错误
### 📸 截图或视频
<!-- ✍ 如果可能的话，附上视频或者截图有助于解决问题 -->
![image](https://github.com/PowerNukkitX/PowerNukkitX/assets/60167194/60baec81-3502-4d27-a4fd-023da3c36907)


### ▶ 复现步骤
<!--- ✍ 详细描述此问题的复现步骤 -->
不清是什么原因
个人推测可能是
1.服务器断电
2.之前安装了科技黎明使用了里面的方块然后又删除了

### ✔ 预期行为
<!-- ✍ 正常情况下应该发生什么 -->
没有报错

### ❌ 实际行为
<!-- ✍ 但是实际上发生了什么不该发生的 -->
报错了

### 📋 Debug information
<!-- 使用 'debugpaste upload' 和 '' 指令并粘贴结果至此 -->
[debugpaste-2023-07-02T15.30.33.172+0800.zip](https://github.com/PowerNukkitX/PowerNukkitX/files/11928916/debugpaste-2023-07-02T15.30.33.172%2B0800.zip)
[20230702153153_NullPointerException.md](https://github.com/PowerNukkitX/PowerNukkitX/files/11928918/20230702153153_NullPointerException.md)

### 💢 错误堆栈或崩溃日志
<!-- ✍ 最好使用 https://hastebin.com -->
15:26:47 [ERROR] 卸载区块时发生错误：cn.nukkit.utils.ChunkException: Error saving chunk (32, -85)
cn.nukkit.utils.ChunkException: Error saving chunk (32, -85)
        at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:358) ~[powernukkitx.jar:?]
        at cn.nukkit.level.Level.unloadChunk(Level.java:4237) [powernukkitx.jar:?]
        at cn.nukkit.level.Level.unloadChunk(Level.java:4198) [powernukkitx.jar:?]
        at cn.nukkit.level.Level.unloadChunks(Level.java:4662) [powernukkitx.jar:?]
        at cn.nukkit.level.Level.doGarbageCollection(Level.java:4562) [powernukkitx.jar:?]
        at cn.nukkit.Server.tickProcessor(Server.java:1165) [powernukkitx.jar:?]
        at cn.nukkit.Server.start(Server.java:1133) [powernukkitx.jar:?]
        at cn.nukkit.Server.<init>(Server.java:960) [powernukkitx.jar:?]
        at cn.nukkit.Nukkit.main(Nukkit.java:160) [powernukkitx.jar:?]
Caused by: java.lang.NullPointerException: Cannot invoke "cn.nukkit.block.customblock.CustomBlock.getNamespaceId()" because the return value of "it.unimi.dsi.fastutil.ints.Int2ObjectMap.get(int)" is null
        at cn.nukkit.level.format.anvil.ChunkSection.toNBT(ChunkSection.java:899) ~[powernukkitx.jar:?]
        at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:529) ~[powernukkitx.jar:?]
        at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:172) ~[powernukkitx.jar:?]
        at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:356) ~[powernukkitx.jar:?]
        ... 8 more

### 💬 其他信息
<!-- ✍ 其他可能有助于解决此问题的信息 -->

--- a/src/main/java/cn/nukkit/level/format/anvil/ChunkSection.java
+++ b/src/main/java/cn/nukkit/level/format/anvil/ChunkSection.java
@@ -896,11 +896,11 @@ public class ChunkSection implements cn.nukkit.level.format.ChunkSection, ChunkS
                     var customBlocksIdMap = new CompoundTag();
                     for (IntIterator iterator = usedCustomIds.intIterator(); iterator.hasNext(); ) {
                         int each = iterator.nextInt();
-                        var namespaceId = Block.ID_TO_CUSTOM_BLOCK.get(each).getNamespaceId();
-                        if (namespaceId == null) {
+                        var block = Block.ID_TO_CUSTOM_BLOCK.get(each);
+                        if (block == null) {
                             log.warn(Server.getInstance().getLanguage().tr("nukkit.anvil.save.unknown-custom-block", each));
                         } else {
-                            customBlocksIdMap.putInt(namespaceId, each);
+                            customBlocksIdMap.putInt(block.getNamespaceId(), each);
                         }
                     }
                     storageTag.putCompound("CustomBlocksIdMap", customBlocksIdMap);

